tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs

```diff
@@ -877,17 +877,19 @@ public void PreventChangesInOpenIdConnectOptionsToBeOverlooked()
         {
             // If the number of public properties of OpenIdConnectOptions changes,
             // then, the PopulateOpenIdOptionsFromMergedOptions method
-            // needs to be updated. For this uncomment the 2 lines below, and run the test
+            // needs to be updated. For this uncomment the System.IO.File.WriteAllLines() lines below, and run the test
             // then diff the files to find what are the new properties
             int numberOfProperties = typeof(OpenIdConnectOptions).GetProperties(System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Instance).Length;
```

Comment:
>numberOfProperties [](http://example.com/codeflow?start=16&length=18)

While this works it can be improved, this test will not catch regressions where a property is added and another is removed. For a round test case you need a static definition of the properties expected on each .NET version and to compare against it.

Reply:
Since you have code here that drops the properties in a file, I would recommend using this file to add those arrays to the code.